lib/mixins/enumerable.js

```diff
@@ -2,16 +2,28 @@
 
 var Enumerable = {
   list: {
-    value: function (query) {
-      return this.listAll(query);
+    value: function (query, alt_auth) {
+      return this.listAll(query, alt_auth);
     }
   },
   listAll: {
-    value: function (query) {
-      return this.request({
+    value: function (query, alt_auth) {
+      var request_opts = {
         method: 'get',
         query: query
-      }).then(function (res) {
+      };
+
+      if (alt_auth && alt_auth.token) {
+        request_opts.headers = {
+          'x-auth-token': alt_auth.token
+        };
+      } else if (alt_auth && alt_auth.api_key) {
+        request_opts.headers = {
+          'st2-api-key': alt_auth.api_key
+        };
+      }
+
```

Comment:
@blag If this part of code is duplicated in almost all mixins, could it be a function call in util or other file?

Reply:
Fixed.